frontend/src/components/question-creation/question-elements/SaveQuestionEditorButton.tsx

```diff
@@ -112,17 +114,52 @@ const SaveQuestionEditorButton = ({
     return !emptyEditor;
   };
 
+  const addQuestion = (newQuestionElements: QuestionElement[]) => {
+    setQuestions((prevQuestions) => {
+      return [
+        ...prevQuestions,
+        { id: uuidv4(), elements: newQuestionElements },
+      ];
+    });
+  };
+
+  const updateQuestion = (
+    id: string,
+    updatedQuestionElements: QuestionElement[],
+  ) => {
+    setQuestions((prevQuestions) => {
+      const indexToUpdate = prevQuestions.findIndex(
+        (question) => question.id === id,
+      );
+      return update(prevQuestions, {
+        [indexToUpdate]: {
+          $merge: {
+            elements: updatedQuestionElements,
+          },
+        },
+      });
+    });
+  };
```

Comment:
I think there is a lot of this type of logic in the question editor features and i want to refactor to somewhere common + cleanup code utils. but holding off on this since we're in a hurry to get this pushed out